samples/functions/csharp/simplechat-wpscontext/Function1.cs

```diff
@@ -80,7 +83,25 @@ public static IActionResult Home([HttpTrigger(AuthorizationLevel.Anonymous)] Htt
                 await actions.AddAsync(WebPubSubAction.CreateSendToAllAction(request.Data, request.DataType));
             }
 
-            return new ClientContent("ack").ToString();
+            // Retrieve counter from states.
+            // Input binding has limitation to help users build service required responses. So users have to do this themselves.
+            // See example below of how to get/set connection state correctly from WebPubSubContext.Request.ConnectionContext.Headers["ce-connectionstate"].
+            var states = new CounterState(1);
+            var idle = 0.0;
+            if (wpsReq.Request.ConnectionContext.Headers.TryGetValue("ce-connectionState", out var counterValue))
+            {
+                // Get states.
+                states = JsonConvert.DeserializeObject<CounterState>(Encoding.UTF8.GetString(Convert.FromBase64String(counterValue.SingleOrDefault())));
+                idle = (DateTime.Now - states.Timestamp).TotalSeconds;
+                states.Update();
+            }
+
+            var response = new HttpResponseMessage();
+            response.Content = new StringContent(new ClientContent($"ack, idle: {idle}s, connection message counter: {states.Counter}").ToString());
+            // Set states.
+            response.Headers.Add("ce-connectionState", Convert.ToBase64String(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(states))));
```

Comment:
is using `System.Text.Json.JsonSerializer.SerializeToUtf8Bytes` more straightforward?

Reply:
Looks good. Switched to `System.Text.Json` in this sample.